### PR TITLE
Add logic for OOT Elegy

### PR DESCRIPTION
**Pull request by XenoWars**
This adds the logic for OOT Elegy to the relevant blue switches.

### CODE REVIEW

File: packages/data/src/world/mq/deku_tree_mq.yml [hunk 1]
```diff
@@ -70,6 +70,8 @@
   exits:
     "Deku Tree Lobby": "true"
     "Deku Tree Water Room Back": "(is_child && has_shield) || can_longshot || (can_hookshot && has_iron_boots)"
+  events:
+    MQ_DEKU_WATER_TORCHES: "(is_child && has_shield && can_use_sticks) || (can_use_bow && can_play_elegy)"
```
Comment (ZoeyZolotova) [inline packages/data/src/world/mq/deku_tree_mq.yml:74]: Does this need `|| has_fire` like was removed from line 81?

Comment (ZoeyZolotova) [reply packages/data/src/world/mq/deku_tree_mq.yml:74]: Oh nevermind.

Comment (ZoeyZolotova) [reply packages/data/src/world/mq/deku_tree_mq.yml:74]: Could fire arrow matter here, if you don't have elegy?

File: packages/data/src/world/mq/fire_temple_mq.yml [hunk 2]
```diff
@@ -61,6 +61,7 @@
     "Fire Temple Maze Upper": "can_hookshot && (trick(OOT_HAMMER_WALLS) || has_explosives)"
   locations:
     "MQ Fire Temple Maze Lower Chest": "true"
+    "MQ Fire Temple Maze Side Room Chest": "can_play_elegy"
```
Comment (ZoeyZolotova) [inline packages/data/src/world/mq/fire_temple_mq.yml:64]: Are you normally supposed to bring a crate from Maze Upper for this?

File: packages/data/src/world/oot/jabu_jabu.yml [hunk 3]
```diff
@@ -7,10 +7,10 @@
   dungeon: JJ
   exits:
     "Jabu-Jabu": "true"
-    "Jabu-Jabu Pre-Boss": "event(BIG_OCTO) || (has_hover_boots && trick(OOT_JABU_BOSS_HOVER)) || climb_anywhere || (glitch_megaflip && trick(OOT_JJB_BOXLESS) && can_jump_slash)"
+    "Jabu-Jabu Pre-Boss": "event(BIG_OCTO) || (has_hover_boots && (trick(OOT_JABU_BOSS_HOVER) || can_play_elegy)) || climb_anywhere || (glitch_megaflip && trick(OOT_JJB_BOXLESS) && can_jump_slash)"
```
Comment (ZoeyZolotova) [inline packages/data/src/world/oot/jabu_jabu.yml]: Should we add can_play_elegy to the last part as an alternative to the JJB_BOXLESS trick?

File: packages/data/src/world/mq/deku_tree_mq.yml (hunk 1 above)
Comment (XenoWars) [reply packages/data/src/world/mq/deku_tree_mq.yml:74]: all it does is open the door on the other side of the room, so no

File: packages/data/src/world/mq/fire_temple_mq.yml (hunk 2 above)
Comment (XenoWars) [reply packages/data/src/world/mq/fire_temple_mq.yml:64]: yeah, from upper. that's why it's originally located there.

File: packages/data/src/world/oot/jabu_jabu.yml (hunk 3 above)
Comment (XenoWars) [reply packages/data/src/world/oot/jabu_jabu.yml]: oh, right. probably should.

**Review by ZoeyZolotova — APPROVED** [review]

**Pull request merged**